app/src/main/java/ch/epfl/sdp/appart/AdActivity.java

```diff
@@ -70,12 +81,23 @@ protected void onCreate(Bundle savedInstanceState) {
         mViewModel.getAddress().observe(this, this::updateAddress);
         mViewModel.getPrice().observe(this, this::updatePrice);
         mViewModel.getDescription().observe(this, this::updateDescription);
-        mViewModel.getAdAdvertiserName().observe(this, this::updateAdvertiserName);
+        mViewModel.getAdAdvertiserName().observe(this,
+                this::updateAdvertiserName);
         mViewModel.getAdvertiserId().observe(this, this::updateAdvertiserId);
-        mViewModel.observePanoramasReferences(this, this::updatePanoramasReferences);
+        mViewModel.observePanoramasReferences(this,
+                this::updatePanoramasReferences);
```

Comment:
I think in the refactor we could use this format to observe the fields, the line 87 comes from one of my PR. You inject the object inside the viewmodel as well as the callback, this avoid leaking references to all fields.